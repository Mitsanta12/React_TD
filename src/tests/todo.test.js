import { fireEvent, render, screen } from '@testing-library/react';
import React from 'react';
import renderer from 'react-test-renderer';
import Todo from '../component/todo';

//tester l'affichage correct de la composante Todo 
describe('Todo component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Todo />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  //pour s'assurer que la valeur de todo est mise à jour correctement 
  it('updates todo when input is changed', () => {
    //simule une fonction
    const mock = jest.fn
      render(<Todo />);
      const input = screen.getByTestId('input')
      fireEvent.change(input, {target: {value: {mock}}})
      expect(mock.call.length).toBe(1)

  });

  // vérifie si le composant Todo ajoute correctement un todo à la liste doing lorsque la méthode handleSubmit est appelée
  it('adds todo to doing when handleSubmit is called', () => {
    const mock = jest.fn
      render( <Todo/>)
      const update = screen.getByTestId('input')
      fireEvent.keyPress(update,{ key: {mock}} )
      expect (mock.call.length).toBe(1)    
  });

  //verifie lorsque le checkbox est checker
  it('it"s checked', () => {
    const mock = jest.fn  
      render( <Todo/>)
        const checked = screen.getByTestId('Checkbox')
        fireEvent.click(checked)
        expect (mock.call.length).toBe(1)
  })

  it('delete todo when he is done correctly', () =>{
    const mock = jest.fn  
      render(<Todo/>)
        const deletedone= screen.getByTestId('done')
        fireEvent.click(deletedone)
        expect (mock.call.length).toBe(1)
  })

  it('adds doing when checkbox is called', () => {
    const mock = jest.fn
      render( <Todo/>)
      const done = screen.getByTestId('done')
      fireEvent.keyPress(done,{ key: {mock}} )
      expect (mock.call.length).toBe(1)    
  });
});
