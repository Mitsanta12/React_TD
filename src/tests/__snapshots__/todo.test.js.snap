// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo component renders correctly 1`] = `
Array [
  <div>
    <h1>
      TODO APPLICATION
    </h1>
    <form
      onSubmit={[Function]}
    >
      <input
        data-testId="input"
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder="Add task"
        type="text"
        value=""
      />
    </form>
  </div>,
  <div
    className="container"
    data-testId="Checkbox"
  >
    <div
      className="todo"
    >
      <h3>
        TODO
      </h3>
      <form>
        <ul />
      </form>
    </div>
    <div
      className="done"
      data-testId="done"
    >
      <h3>
        DONE
      </h3>
      <ul />
    </div>
  </div>,
  ")",
]
`;
